### PR TITLE
Issue 303

### PR DESCRIPTION
**Pull request by JensEggers**

### CODE REVIEW

File: test/b7-shared-data-component.js [hunk 1]
```diff
@@ -107,6 +107,22 @@ describe('b7 - shared data component', function () {
...
     });
 
+    it('respects noPublish set option', function (done) {
+      meshInstance.event.data.on('/some/path/testNoPublish', function (data, meta) {
+        delete data._meta;
+        data.should.eql({val: 'must be emitted'});
+        done();
+      }, function (e) {
+        if (e) return done(e);
+        dataComponent.set('/some/path/testNoPublish', {val: 'must not be emitted'}, {noPublish: true}, function (e) {
```
Comment (JohanvdWest) [inline test/b7-shared-data-component.js]: How do you know this one didn't fire?

Comment (JensEggers) [reply test/b7-shared-data-component.js]: there would still be an exception - but I can improve on this

Comment (JensEggers) [reply test/b7-shared-data-component.js]: made the test wait for a second before exiting; if I force the incorrect value to be emitted by setting noPublish to false, the test fails as expected (which it also did before)

**Comment by JensEggers** [comment]
merging in spite of intermittent failure on travis since that is unrelated (https://github.com/happner/happner/issues/305)

**Pull request merged**